### PR TITLE
WT-10276 Improve RTS test coverage

### PR DESCRIPTION
**Pull request by korteland**
Of the four scenarios in the ticket:
1. We don't want a test that runs long enough to trigger this.
2. This was being tested, but not by an RTS test.
3. This was also being tested, but the logging was done under the wrong subsystem, so fix it.
4. This is almost all of the PR. It needed some supporting infra in `wttest.py`, in particular the idea of a custom log validator and `ignoreTeardownLogs`. This latter part seems a little hacky to me, but I'm not sure how to improve it - perhaps a getter/setter? Feedback welcome here.

The other possibly-controversial part is the lack of Windows support - we delete the file while WiredTiger has it open, and this doesn't work nicely with the Windows model of "only one process opens the file".

### CODE REVIEW

File: test/suite/test_rollback_to_stable42.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+        # RTS runs as part of shutdown, but it won't succeed since we
+        # remove the file that needs rollback.
+        self.ignoreTearDownLogs = True
+
+        uri = 'table:test_rollback_to_stable42'
+        nrows = 1000
+
+        if self.value_format == '8t':
+            value_a = 97
```
Comment (ddanderson) [inline test/suite/test_rollback_to_stable42.py:96]: `value_a` is unused.

**Review by ddanderson — APPROVED** [review]
LGTM - one minor nit.

File: test/suite/test_rollback_to_stable42.py (hunk 1 above)
Comment (korteland) [reply test/suite/test_rollback_to_stable42.py]: Oops - thanks! Fixed in 8125291.

**Review by agorrod — APPROVED** [review]
lgtm

**Comment by korteland** [comment]
evergreen merge

**Pull request merged**